perses/annihilation/new_relative.py

```diff
@@ -0,0 +1,1213 @@
+import simtk.openmm as openmm
+import simtk.openmm.app as app
+import simtk.unit as unit
+import mdtraj as md
+import numpy as np
+import copy
+import enum
+
+ONE_4PI_EPS0 = 138.935456
+InteractionGroup = enum.Enum("InteractionGroup", ['unique_old', 'unique_new', 'core', 'environment'])
+
+class HybridTopologyFactory(object):
+    """
+    This class generates a hybrid topology based on a perses topology proposal. This class treats atoms
+    in the resulting hybrid system as being from one of four classes:
+
+    unique_old_atom : these atoms are not mapped and only present in the old system. Their interactions will be on for
+        lambda=0, off for lambda=1
+    unique_new_atom : these atoms are not mapped and only present in the new system. Their interactions will be off
+        for lambda=0, on for lambda=1
+    core_atom : these atoms are mapped, and are part of a residue that is changing. Their interactions will be those
+        corresponding to the old system at lambda=0, and those corresponding to the new system at lambda=1
+    environment_atom : these atoms are mapped, and are not part of a changing residue. Their interactions are always
+        on and are alchemically unmodified.
+    """
+
+    def __init__(self, topology_proposal, current_positions, new_positions):
+        """
+        Initialize the Hybrid topology factory.
+
+        Parameters
+        ----------
+        topology_proposal : perses.rjmc.topology_proposal.TopologyProposal object
+            TopologyProposal object rendered by the ProposalEngine
+        current_positions : [n,3] np.ndarray of float
+            The positions of the "old system"
+        new_positions : [m,3] np.ndarray of float
+            The positions of the "new system"
+        """
+        self._topology_proposal = topology_proposal
+        self._old_system = copy.deepcopy(topology_proposal.old_system)
+        self._new_system = copy.deepcopy(topology_proposal.new_system)
+        self._old_to_hybrid_map = {}
+        self._new_to_hybrid_map = {}
+        self._hybrid_system_forces = dict()
+        self._old_positions = current_positions
+        self._new_positions = new_positions
+
+        self.softcore_alpha=0.5
+        self.softcore_beta=12*unit.angstrom**2
+
+        #prepare dicts of forces, which will be useful later
+        self._old_system_forces = {type(force).__name__ : force for force in self._old_system.getForces()}
+        self._new_system_forces = {type(force).__name__ : force for force in self._new_system.getForces()}
+
+        #get and store the nonbonded method from the system:
+        self._nonbonded_method = self._old_system_forces['NonbondedForce'].getNonbondedMethod()
+
+        #start by creating an empty system. This will become the hybrid system.
+        self._hybrid_system = openmm.System()
```

Comment:
I realize what you are saying, but I think it's a tradeoff. I actually think that it's easier to make a mistake assigning parameters to particles (as well as leading to more confusing code, as the previous hybrid implementation) if we copy a system and then modify it. On the other hand, it's easy to forget the barostat, etc. if we create a new system. However, it's much easier (in my view) to make sure that we remember those things (barostat, virtual sites, etc.) than to ensure that we have deleted parameters correctly. 

With regard to weird forces, I think we should just throw an exception (or maybe a warning). I don't think we want to try to handle unknown forces.